modules/calib3d/src/checkchessboard.cpp

```diff
@@ -55,15 +55,12 @@ static void icvGetQuadrangleHypotheses(const std::vector<std::vector< cv::Point
     const float max_aspect_ratio = 3.0f;
     const float min_box_size = 10.0f;
 
-    typedef std::vector< std::vector< cv::Point > >::const_iterator iter_t;
-    iter_t i;
-    for (i = contours.begin(); i != contours.end(); ++i)
+    for (size_t i = 0; i < contours.size(); ++i)
     {
-        const iter_t::difference_type idx = i - contours.begin();
```

Comment:
Raw pointers have no ::difference_type member typedef (due to not having members at all).
One could use std::iterator_traits to get one, by I have simplified to code instead.